crates/interledger-api/src/routes/accounts.rs

```diff
@@ -409,3 +449,59 @@ where
         .or(post_payments)
         .boxed()
 }
+
+fn get_address_from_parent_and_update_routes<S, A, T>(
+    mut service: S,
+    parent: A,
+    store: T,
+) -> impl Future<Item = (), Error = ()>
+where
+    S: OutgoingService<A> + Clone + Send + Sync + 'static,
+    A: CcpRoutingAccount + Clone + Send + Sync + 'static,
+    T: AddressStore + Clone + Send + Sync + 'static,
+{
+    let prepare = IldcpRequest {}.to_prepare();
+    service
+        .send_request(OutgoingRequest {
+            from: parent.clone(), // Does not matter what we put here, they will get the account from the HTTP/BTP credentials
+            to: parent.clone(),
+            prepare,
+            original_amount: 0,
+        })
+        .map_err(|err| error!("Error getting ILDCP info: {:?}", err))
+        .and_then(|fulfill| {
+            let response = IldcpResponse::try_from(fulfill.into_data().freeze()).map_err(|err| {
+                error!(
+                    "Unable to parse ILDCP response from fulfill packet: {:?}",
+                    err
+                );
+            });
+            debug!("Got ILDCP response: {:?}", response);
+            let ilp_address = match response {
+                Ok(info) => info.ilp_address(),
+                Err(_) => return err(()),
+            };
+            ok(ilp_address)
+        })
+        .and_then(move |ilp_address| store.set_ilp_address(ilp_address))
+        .and_then(move |_| {
+            // Get the parent's routes for us
+            let prepare = RouteControlRequest {
```

Comment:
Instead of constructing a `RouteControlRequest` here, what do you think about having a handle to the `CcpServer` to trigger it to send the request? If that's a lot more complicated, we can keep this as is

Reply:
We could but I'd prefer not. CcpRouteManager is generic over `Incoming`, `Outgoing` services and a `Store`, and that'd require adding extra generic parameters on the API. 

The 'benefit" would be being more explicit on the types of outgoing requests that we can make. I don't think that's really beneficial, since in the future we might want to perform other actions and send other types of requests, than just CCP requests via the outgoing handler.

Reply:
Sounds good for now. Mind adding a TODO to make a note about this?
```suggestion
            // TODO we may want to make this trigger the CcpRouteManager to request
            // routes instead of sending a RouteControlRequest here
            let prepare = RouteControlRequest {
```